### PR TITLE
Add workflow to upload BPF performance results

### PR DESCRIPTION
**Pull request by Alan-Jowett**
## Description

Add workflow to upload BPF performance results to bpf-performance-benchmarks.postgres.database.azure.com.

## Testing

CI/CD

## Documentation

No.

## Installation

No.

### CODE REVIEW

File: .github/workflows/cicd.yml [hunk 1]
```diff
@@ -136,7 +136,7 @@ jobs:
     # Always run this job.
     # Only run this on repos that have self-host runners.
     needs: regular
-    if: github.repository == 'microsoft/ebpf-for-windows'
+    if: github.repository == 'microsoft/ebpf-for-windows' && (github.event_name == 'schedule' || github.event_name == 'pull_request' || github.event_name == 'push' || github.event_name == 'merge_group')
```
Comment (rectified95) [inline .github/workflows/cicd.yml:139]: is this condition equivalent to 'run always except on manual workflow start'?
If so, can you simplify this if statement?

Also, are the perf tests the only ones we would ever run manually?

Comment (rectified95) [reply .github/workflows/cicd.yml:139]: Ah, seeing below it's only the performance_with_profile that is supposed to be run manually, right?

File: .github/workflows/upload-results.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    - name: Download performance result artifacts
+      uses: actions/download-artifact@v2
+      with:
+        name: Test-Logs-${{inputs.result_artifact}}
+        path: ${{github.workspace}}/results_artifact
+
+    - name: Gather CSV results into results directory
+      run: |
+        mkdir -p ${{github.workspace}}/results/${{github.sha}}
+        curl https://raw.githubusercontent.com/microsoft/bpf_performance/1a50dae54c0c1d8b7f73331936e69a7154b3f659/scripts/gather_csv.sh > gather_csv.sh
```
Comment (rectified95) [inline .github/workflows/upload-results.yml:31]: I think in another PR you were parametrizing the download to get the latest bits - is it easy to do it here too?

**Review by rectified95 — DISMISSED** [review]

File: .github/workflows/upload-results.yml (hunk 2 above)
Comment (Alan-Jowett) [reply .github/workflows/upload-results.yml:31]: The goal here is to make sure the script we get hasn't been altered.

Comment (rectified95) [reply .github/workflows/upload-results.yml:31]: Sure, was just thinking in terms of not needing to update this line if there are changes to the script. Not blocking here

File: .github/workflows/cicd.yml (hunk 1 above)
Comment (rectified95) [reply .github/workflows/cicd.yml:139]: 1. For my information - can this if-statement be simplified as 'github.event_name != dispatch'?
2. Wondering if we should create a new workflow for this - I see a possibility we want to manually trigger the CICD and have everything run.

File: .github/workflows/upload-results.yml (hunk 2 above)
Comment (Alan-Jowett) [reply .github/workflows/upload-results.yml:31]: That's intentional. If the script changes, then someone should review it before picking up the new version to make sure it's not malicious.

File: .github/workflows/cicd.yml (hunk 1 above)
Comment (Alan-Jowett) [reply .github/workflows/cicd.yml:139]: Maybe. Can we do this as a separate PR?

Comment (rectified95) [reply .github/workflows/cicd.yml:139]: Sure

**Review by rectified95 — DISMISSED** [review]

**Review by shankarseal — APPROVED** [review]
rename upload_results to upload_perf_results.yml.

File: .github/workflows/cicd.yml [hunk 3]
```diff
@@ -498,5 +501,13 @@ jobs:
       test_command: .\execute_ebpf_cicd_tests.ps1 -TestMode "Performance" -Options @("CaptureProfile")
       post_test: .\cleanup_ebpf_cicd_tests.ps1
       build_artifact: Build-x64
-      environment: ebpf_cicd_perf
```
Comment (rectified95) [inline .github/workflows/cicd.yml:501]: Did you delete this on purpose?

File: .github/workflows/cicd.yml [hunk 4]
```diff
@@ -30,7 +30,7 @@ concurrency:
   cancel-in-progress: true
 
 permissions:
-  contents: read
+  contents: write # Required to upload performance results.
```
Comment (rectified95) [inline .github/workflows/cicd.yml:33]: This is another reason I'd suggest extracting the perf job into a separate GH workflow after this change.

Comment (Alan-Jowett) [reply .github/workflows/cicd.yml]: Removed.

**Review by shankarseal — APPROVED** [review]

**Review by rectified95 — APPROVED** [review]
LGTM

**Pull request merged**